core/src/test/scala/org/apache/spark/util/collection/OpenHashSetSuite.scala

```diff
@@ -255,4 +255,17 @@ class OpenHashSetSuite extends SparkFunSuite with Matchers {
     val set = new OpenHashSet[Long](0)
     assert(set.size === 0)
   }
+
+  test("support for more than 12M items") {
+    val cnt = 12000000 // 12M
+    val set = new OpenHashSet[Int](cnt)
+    for (i <- 0 until cnt) {
+      set.add(i)
+      assert(set.contains(i))
+
+      val pos1 = set.addWithoutResize(i) & OpenHashSet.POSITION_MASK
+      val pos2 = set.getPos(i)
```

Comment:
Thank you for doing this, @viirya .

Shall we switch the line 267 and 266 because `set.addWithoutResize(i)` has side-effects like changing the underlying structure. Although the situation will not happen because we called `addWithoutResize` inside `add` at line 263. But, we had better get the comparison target value from the read-only function first.